torchtune/modules/position_embeddings.py

```diff
@@ -42,20 +42,21 @@ def __init__(
         self.dim = dim
         self.base = base
         self.max_seq_len = max_seq_len
-        self._rope_init()
+        theta = self.get_theta()
+        self.register_buffer("theta", theta, persistent=False)
+        self.build_rope_cache(self.max_seq_len)
 
     # We need to explicitly define reset_parameters for FSDP initialization, see
     # https://github.com/pytorch/pytorch/blob/797d4fbdf423dd9320ebe383fb57ffb1135c4a99/torch/distributed/fsdp/_init_utils.py#L885
     def reset_parameters(self):
-        self._rope_init()
```

Comment:
prefer `copy_` over registering buffer again since this is resetting

Reply:
Why not make `get_theta()` just register the buffer again?

I did not think that you can `copy_` into meta-device tensor 🤔 curious how this works

Reply:
> just register the buffer again?

registering again is fine. I called `module.to_empy(self._deivce)` before calling `module.reset_parameters()`. By default `get_theta()`  creates tensors on cpu. It's more like a stylish thing where I followed fsdp unit test

Reply:
Hmm. I am not sure we need to call `module.to_empty()`. That would create an extra sharded copy of the parameters on GPU when there is also one in the sharded state dict?

Reply:
The point of `load_state_dict(assign=True)` is that the `module` can still have meta-device parameters when we are loading, as long as the state dict itself has GPU parameters.

Reply:
oh, I only call `module.to_empty()` if module is on meta device. but again, it's stylish thing. another way is simply call `_rope_init()` with `self._device` ctx

Reply:
got you. will call `_rope_init()` instead